### PR TITLE
.net app for CBL2.0

### PR DESCRIPTION
**Pull request by hrajput89**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- .net App for 2.0

### CODE REVIEW

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer/EncryptionKeyMethods.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        //                                    [NotNull] IReadOnlyDictionary<string, object> postBody,
+        //                                    [NotNull] HttpListenerResponse response)
+        //{
+        //    var key = (byte[])postBody["key"];
+        //    var password = postBody["password"].ToString();
+
+        //    if (password != null){
+        //        response.WriteBody(MemoryMap.New<EncryptionKey>(password));
+        //    }
+        //    else if (key != null){
```
Comment (sridevi-15) [inline CBLClient/Apps/CBLTestServer-Dotnet/TestServer/EncryptionKeyMethods.cs]: we do not pass key , we pass password. Get encryption key using password

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ReplicatorConfigurationMethods.cs [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+                    });
+                }
+                else
+                {
+                    throw new ArgumentException("Invalid value for replication_type");
+                }
+            });
+        }
+
+        //public static void Copy([NotNull] NameValueCollection args,
```
Comment (sridevi-15) [inline CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ReplicatorConfigurationMethods.cs]: i do not see this method in iOS , is it required for .net?

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ReplicatorConfigurationMethods.cs [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+            {
+                replType = ReplicatorType.PushAndPull;
+            }
+            With<ReplicatorConfiguration>(postBody, "configuration", repConf =>
+            {
+                repConf.ReplicatorType = replType;
+            });
+        }
+
+        //public static void setPinnedServerCertificate([NotNull] NameValueCollection args,
```
Comment (sridevi-15) [inline CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ReplicatorConfigurationMethods.cs]: we need this API as it exists for iOS and Android

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer/SessionAuthenticationMethods.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        }
+
+        public static void GetCookieName([NotNull] NameValueCollection args,
+                        [NotNull] IReadOnlyDictionary<string, object> postBody,
+                        [NotNull] HttpListenerResponse response)
+        {
+            With<SessionAuthenticator>(postBody, "session", si => response.WriteBody(si.CookieName));
+        }
+
+        //public static void GetExpires([NotNull] NameValueCollection args,
```
Comment (sridevi-15) [inline CBLClient/Apps/CBLTestServer-Dotnet/TestServer/SessionAuthenticationMethods.cs]: is this needed? I do not see this API on iOS and Android?

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer/TestServer.cs [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+using JetBrains.Annotations;
+
+namespace Couchbase.Lite.Testing
+{
+    public sealed class TestServer
+    {
+        #region Constants
+
+        private const ushort Port = 55555;
```
Comment (sridevi-15) [inline CBLClient/Apps/CBLTestServer-Dotnet/TestServer/TestServer.cs]: change the port to 8080 to have consistent with iOS and Android

File: CBLClient/Apps/CBLTestServer-Dotnet/TestClient/Orchestration/GatewayConfig.json [hunk 6]
```diff
@@ -0,0 +1,64 @@
+{
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestClient/Orchestration/GatewayConfig.json]: deleting it, as it was just for the experimentation purpose.

File: CBLClient/Apps/CBLTestServer-Dotnet/TestClient/Program.cs [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+using Newtonsoft.Json;
+
+using TestClient.Orchestration;
+
+namespace TestClient
+{
+    class Program
+    {
+        public const string ServerUrl = "http://192.168.1.5:55555/";
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestClient/Program.cs]: deleting it, as it was just for the experimentation purpose.

File: CBLClient/Apps/CBLTestServer-Dotnet/TestClient/Program.cs [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        public const string SyncGatewayOrchestrationUrl = "http://192.168.1.2:55555/";
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestClient/Program.cs]: deleting it, as it was just for the experimentation purpose.

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer.Android/obj/Debug/build.props [hunk 9]
```diff
@@ -0,0 +1,22 @@
+BundleAssemblies=False
+AotAssemblies=False
+AndroidAotMode=
+ExplicitCrunch=False
+EnableProguard=
+UseSharedRuntime=True
+EmbedAssembliesIntoApk=False
+AndroidLinkMode=None
+AndroidLinkSkip=
+AndroidSdkBuildToolsVersion=23.0.0
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestServer.Android/obj/Debug/build.props]: do we need to make any changes to code?

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer.Android/obj/Debug/devices.cache [hunk 10]
```diff
@@ -0,0 +1,7 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Devices>
+  <Device id="192.168.60.101:5555">
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestServer.Android/obj/Debug/devices.cache]: this file is generated by VS. I'll remove it

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer.NetCore/QueryMethods.cs [hunk 11]
```diff
@@ -0,0 +1,147 @@
+////
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestServer.NetCore/QueryMethods.cs]: I'll remove this one too.

[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestServer.NetCore/QueryMethods.cs]: experimental file, but need changes to program.cs

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer.NetCore/OrchestrationMethods.cs [hunk 12]
```diff
@@ -0,0 +1,150 @@
+//
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestServer.NetCore/OrchestrationMethods.cs]: experimental file, but need changes to program.cs

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ConflictMethods.cs [hunk 13]
```diff
@@ -0,0 +1,20 @@
+using System;
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ConflictMethods.cs]: Will remove it.

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer/DatabaseMethods.cs [hunk 14]
```diff
@@ -0,0 +1,511 @@
...
+            var handle = postBody[key].ToString();
+            var db = MemoryMap.Get<T>(handle);
+            await action(db);
+        }
+
+        #endregion
+
+        #region Internal Methods
+
+        //internal static void DatabaseAddChangeListener([NotNull] NameValueCollection args,
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestServer/DatabaseMethods.cs]: Should I enable it now?

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer/DatabaseMethods.cs [hunk 15]
```diff
@@ -0,0 +1,511 @@
...
+                    } catch (InvalidOperationException e)
+                    {
+                        Console.WriteLine("Error deleting DB" + e.Message);
+                        response.WriteBody(e.Message);
+                    }
+                });
+            });
+        }
+
+        //internal static void DatabaseRemoveChangeListener([NotNull] NameValueCollection args,
```
[the remark this answers is not in the document]
Comment (hrajput89) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestServer/DatabaseMethods.cs]: Should I enable it now?

**Pull request closed without merging**